### PR TITLE
Configure fota with defines instead of patching

### PR DESCRIPTION
**Pull request by kivaisan**
Use defines to configure FOTA locations.

### CODE REVIEW

**Comment by kivaisan** [comment]
@JanneKiiskila @esajaa @TapaniPuhakka @petedyerarm please review. I didn't do the upper level cmake here yet as it would require more testing on HWs I don't have.

**Review by esajaa — APPROVED** [review]
LGTM. Please create a jira ticket about the upper-level common configuration so that it isn't forgotten

**Review by JanneKiiskila — APPROVED** [review]

**Comment by kivaisan** [comment]
IOTABBGW-189 created to implemtn common cmake configuration file.

File: recipes-connectivity/mbed-edge-core/files/uz/target.cmake [hunk 1]
```diff
@@ -8,6 +8,12 @@ SET (BIND_TO_ALL_INTERFACES 0)
 SET (PAL_UPDATE_FIRMWARE_DIR "\"/mnt/cache/firmware\"")
 SET (ARM_UC_SOCKET_TIMEOUT_MS 300000)
 
+add_definitions(
```
Comment (TapaniPuhakka) [inline recipes-connectivity/mbed-edge-core/files/uz/target.cmake:11]: should these be under if (${FOTA_ENABLE}) (see line 28) ?

File: recipes-connectivity/mbed-edge-core/files/imx8mmevk/target.cmake [hunk 2]
```diff
@@ -8,6 +8,12 @@ SET (BIND_TO_ALL_INTERFACES 0)
 SET (PAL_UPDATE_FIRMWARE_DIR "\"/mnt/cache/firmware\"")
 SET (ARM_UC_SOCKET_TIMEOUT_MS 300000)
 
+add_definitions(
```
Comment (TapaniPuhakka) [inline recipes-connectivity/mbed-edge-core/files/imx8mmevk/target.cmake:11]: likely also should be flagged with fota enabled

Comment (TapaniPuhakka) [reply recipes-connectivity/mbed-edge-core/files/imx8mmevk/target.cmake:11]: as done with recipes-connectivity/mbed-edge-core/files/uz/target.cmake

File: recipes-connectivity/mbed-edge-core/files/0003-fix_libwebsockets_nondebug.manual_patch [hunk 3]
```diff
@@ -1,13 +0,0 @@
-diff --git a/lib/libwebsockets/libwebsockets/CMakeLists.txt b/lib/libwebsockets/libwebsockets/CMakeLists.txt.txt
```
Comment (TapaniPuhakka) [inline recipes-connectivity/mbed-edge-core/files/0003-fix_libwebsockets_nondebug.manual_patch:1]: why was this removed?

**Comment by TapaniPuhakka** [comment]
additionally I think we should have small change in the fota itself to have this configurable.
https://github.com/PelionIoT/meta-mbed-edge/blob/2ae666666d4afba6596e4f80c6a3048b45e92f32/recipes-connectivity/mbed-edge-core/files/0006-fota-callback.patch

**Comment by TapaniPuhakka** [comment]
> additionally I think we should have small change in the fota itself to have this configurable.
> https://github.com/PelionIoT/meta-mbed-edge/blob/2ae666666d4afba6596e4f80c6a3048b45e92f32/recipes-connectivity/mbed-edge-core/files/0006-fota-callback.patch

actually can't even find fota/fota_install_callback.c . has that been removed already?

**Comment by petedyerarm** [comment]
It is here:
https://github.com/PelionIoT/mbed-edge/blob/master/fota/fota_install_callback.c

**Comment by TapaniPuhakka** [comment]
> It is here:
> https://github.com/PelionIoT/mbed-edge/blob/master/fota/fota_install_callback.c

ok so edge has different fota than mbed-cloud-client ?

**Comment by TapaniPuhakka** [comment]
> > It is here:
> > https://github.com/PelionIoT/mbed-edge/blob/master/fota/fota_install_callback.c
> 
> ok so edge has different fota than mbed-cloud-client ?

ah ok, I see so for linux there needs to be something that implements that fota_app_on_install_candidate functions on the "app" side. No default implementation even provided. So client update has dependency to this single from from the mbed-edge and then this version of this file has dependency to our meta-mbed-edge to provide patch to modify that file.

**Comment by petedyerarm** [comment]
mbed-edge contains mbed-cloud-client

https://github.com/PelionIoT/mbed-edge/tree/master/lib/mbed-cloud-client

File: recipes-connectivity/mbed-edge-core/files/0003-fix_libwebsockets_nondebug.manual_patch (hunk 3 above)
Comment (kivaisan) [reply recipes-connectivity/mbed-edge-core/files/0003-fix_libwebsockets_nondebug.manual_patch:1]: It was an unused file. There was still this old leftover comment for it: `# Temp fix for libwebsockets non-debug build` . I removed that as well.

File: recipes-connectivity/mbed-edge-core/files/uz/target.cmake (hunk 1 above)
Comment (kivaisan) [reply recipes-connectivity/mbed-edge-core/files/uz/target.cmake]: Fixed.

File: recipes-connectivity/mbed-edge-core/files/imx8mmevk/target.cmake (hunk 2 above)
Comment (kivaisan) [reply recipes-connectivity/mbed-edge-core/files/imx8mmevk/target.cmake]: Fixed

File: recipes-connectivity/mbed-edge-core/files/0003-fix_libwebsockets_nondebug.manual_patch (hunk 3 above)
Comment (TapaniPuhakka) [reply recipes-connectivity/mbed-edge-core/files/0003-fix_libwebsockets_nondebug.manual_patch:1]: ok good

**Comment by kivaisan** [comment]
Could this be merged?

**Comment by petedyerarm** [comment]
It has conflicts that need to be resolved.

**Comment by JanneKiiskila** [comment]
Rebase needed, we have a conflict @kivaisan .

File: recipes-connectivity/mbed-edge-core/mbed-edge-core.inc [hunk 4]
```diff
@@ -19,8 +19,7 @@ SRC_URI = "git://github.com/ARMmbed/mbed-edge.git \
            file://0001-disable-Doxygen.patch \
            file://0001-parsec-se-driver-should-be-compiled-separately-in-Yo.patch \
            file://0002-fix-libevent-build-with-CMake-in-Yocto.patch \
-           file://0003-fix_libwebsockets_nondebug.manual_patch \
```
Comment (JanneKiiskila) [inline recipes-connectivity/mbed-edge-core/mbed-edge-core.inc:22]: Is this file (0003) really left out with some good reason? @kivaisan ?

**Review by JanneKiiskila — CHANGES_REQUESTED** [review]
Explanation needed why patch https://github.com/PelionIoT/meta-mbed-edge/blob/master/recipes-connectivity/mbed-edge-core/files/0003-fix_libwebsockets_nondebug.manual_patch is being left out. That has nothing to do with FOTA, AFAIK.

File: recipes-connectivity/mbed-edge-core/mbed-edge-core.inc (hunk 4 above)
Comment (petedyerarm) [reply recipes-connectivity/mbed-edge-core/mbed-edge-core.inc:22]: It no longer gets applied, but the reference to is was not removed. This is a tidy-up.

**Comment by kivaisan** [comment]
> Explanation needed why patch https://github.com/PelionIoT/meta-mbed-edge/blob/master/recipes-connectivity/mbed-edge-core/files/0003-fix_libwebsockets_nondebug.manual_patch is being left out. That has nothing to do with FOTA, AFAIK.

Explained above. See https://github.com/PelionIoT/meta-mbed-edge/pull/52#discussion_r617202444

File: recipes-connectivity/mbed-edge-core/mbed-edge-core.inc (hunk 4 above)
Comment (kivaisan) [reply recipes-connectivity/mbed-edge-core/mbed-edge-core.inc:22]: Explained above. See https://github.com/PelionIoT/meta-mbed-edge/pull/52#discussion_r617202444

**Comment by kivaisan** [comment]
Rebased.

File: recipes-connectivity/mbed-edge-core/files/0003-fix_libwebsockets_nondebug.manual_patch (hunk 3 above)
Comment (JanneKiiskila) [reply recipes-connectivity/mbed-edge-core/files/0003-fix_libwebsockets_nondebug.manual_patch:1]: And we confirmed that the debug build works even w/o this patch @kivaisan ?

Comment (kivaisan) [reply recipes-connectivity/mbed-edge-core/files/0003-fix_libwebsockets_nondebug.manual_patch:1]: This was for **non-debug** builds and that is our default build type: https://github.com/PelionIoT/manifest-lmp-pelion-edge/blob/81c0646badb3f8f0132a14f78f604c978131d793/conf/local.conf#L80

I haven't tried debug builds recently.

**Pull request merged**